base/stream.jl

```diff
@@ -726,93 +725,54 @@ end
 #    finish_read(state...)
 #end
 
-macro uv_write(n,call)
+macro uv_write(stream, data, len)
     esc(quote
         check_open(s)
-        uvw = c_malloc(_sizeof_uv_write+$(n))
-        err = $call
+        uvw = c_malloc(_sizeof_uv_write)
+        uv_req_set_data(uvw,C_NULL)
+        err = ccall(:jl_uv_write,
+                    Int32,
+                    (Ptr{Void}, Ptr{Void}, UInt, Ptr{Void}, Ptr{Void}),
+                    handle($(stream)), $(data), $(len), uvw,
+                    uv_jl_writecb_task::Ptr{Void})
         if err < 0
             c_free(uvw)
             uv_error("write", err)
         end
+        ct = current_task()
+        uv_req_set_data(uvw,ct)
+        ct.state = :waiting
+        stream_wait(ct)
+        c_free(uvw)
```

Comment:
although this seems to be less InterruptException-safe than it was before. perhaps also need to move the JL_SIGATOMIC into this function, since the code it was protecting also moved here